modules/auth/oauth2/oauth2.go

```diff
@@ -41,13 +40,10 @@ type CustomURLMapping struct {
 }
 
 // Init initialize the setup of the OAuth2 library
-func Init() {
-	sessionDir := filepath.Join(setting.AppDataPath, "sessions", "oauth2")
-	if err := os.MkdirAll(sessionDir, 0700); err != nil {
-		log.Fatal(4, "Fail to create dir %s: %v", sessionDir, err)
-	}
-
-	store := sessions.NewFilesystemStore(sessionDir, []byte(sessionUsersStoreKey))
+func Init(x *xorm.Engine) {
+	store := xormstore.NewOptions(x, xormstore.Options{
+		TableName: "oauth2_sessions",
```

Comment:
IMHO we should use `oauth2_session` instead of `oauth2_sessions` because we currently use the singular for table names.